### ISSUE TITLE
Microsoft.Azure.Functions.Worker.Sdk 1.17.0 breaks Durable Functions alternate backends

### ISSUE THREAD

**Issue by cgillum** [Azure/azure-functions-dotnet-worker#2307]
### Description

As reported in https://github.com/microsoft/durabletask-netherite/issues/350, it seems there's a regression in 1.17.0 where the Durable Functions alternate backend assemblies are not being loaded when the function host starts up.

I also tested this using the MSSQL backend and got the same result. The error users see is the following:

> Microsoft.Azure.WebJobs.Extensions.DurableTask: Storage provider type (mssql) was not found. Available storage providers: AzureStorage.

(For Netherite, you will see "netherite" instead of "mssql" in the error message)

This problem does not occur in the 1.16.x versions of Microsoft.Azure.Functions.Worker.Sdk, so this appears to be a regression. I noticed there are several other reports of problems with 1.17 so perhaps they are all related.

/cc @fabiocav 

### Steps to reproduce

Create a Durable Function app that uses one of the alternate backends. Here's an example csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
    <OutputType>Exe</OutputType>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Worker" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask.SqlServer" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" />
    <PackageReference Include="Microsoft.DurableTask.Generators" OutputItemType="Analyzer" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <Using Include="System.Threading.ExecutionContext" Alias="ExecutionContext" />
  </ItemGroup>
</Project>
```

The package versions are as follows:

```xml
<PackageVersion Include="Microsoft.Azure.Functions.Worker" Version="1.20.1" />
<PackageVersion Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" Version="1.0.0" />
<PackageVersion Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.1.1" />
<PackageVersion Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask.SqlServer" Version="1.2.2" />
<PackageVersion Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
<PackageVersion Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.0" />
```

And here's an example host.json:

```
{
  "version": "2.0",
  "extensions": {
    "durableTask": {
      "storageProvider": {
        "type": "mssql",
        "connectionStringName": "SQLDB_Connection"
      }
    }
  }
}
```

**Comment by mikegoatly** [comment]
As per the comments in https://github.com/microsoft/durabletask-netherite/issues/350 it looks like this has been resolved in v1.17.2 of Microsoft.Azure.Functions.Worker.Sdk

**Comment by kshyju** [comment]
This issue was resolved with the latest version of the package 1.17.2
